File: React/pu-project/src/Types/Get/CageDto.ts
export type CageDto = {
  id: number;
  createdOn: string;
  lastModifiedOn: string;
  createdBy: string;
  lastModifiedBy: string;
  animals: CageAnimalDto[];
  events: CageEventDto[];
  capacity: number;
  area: number;
  name: string;
  description: string;
  imageUrl: string;
  location: string;
  rating: number;
  type: string;
};

type CageEventDto = {
  id: number;
  createdOn: string;
  lastModifiedOn: string;
  createdBy: string;
  lastModifiedBy: string;
  cageId: number;
  cage: string;
  type: string;
  description: string;
};

type CageAnimalDto = {
  id: number;
  createdOn: string;
  lastModifiedOn: string;
  createdBy: string;
  lastModifiedBy: string;
  description: string;
  name: string;
  imageUrl: string;
  kingdomType: string;
  age: number;
  gender: string;
  price: number;
  cageId: number;
  cage: string;
  numberOfFoods: number;
  foods: string[];
};

File: ZooErp/Models/CageDto.cs
using System;
namespace ZooErp.Models
{
	public class CageDto: BaseModelDto
	{
        public IEnumerable<AnimalDto> Animals { get; set; }

        public IEnumerable<EventDto> Events { get; set; }

        public int Capacity { get; set; }

        public double Area { get; set; }

        public string Name { get; set; }

        public string Description { get; set; }

        public string ImageUrl { get; set; }

        public string Location { get; set; }

        public float Rating { get; set; }

        public string Type { get; set; }

        public int TypeId { get; set; }
    }
}

